File: rb/features.rb
# TemporaryEmail SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/test_feature'


module TemporaryEmailFeatures
  def self.make_feature(name)
    case name
    when "base"
      TemporaryEmailBaseFeature.new
    when "test"
      TemporaryEmailTestFeature.new
    else
      TemporaryEmailBaseFeature.new
    end
  end
end

File: py/features.py
# TemporaryEmail SDK feature factory

from feature.base_feature import TemporaryEmailBaseFeature
from feature.test_feature import TemporaryEmailTestFeature


def _make_feature(name):
    features = {
        "base": lambda: TemporaryEmailBaseFeature(),
        "test": lambda: TemporaryEmailTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
